refactor(app): use create-react-app for client & merge electron + server

JPSchellenberg opened the pull request.

JPSchellenberg committed:
feat(server): update server

sr258 commented on package.json:
```diff
@@ -89,6 +92,7 @@
         "@sentry/types": "5.29.0",
         "@sentry/webpack-plugin": "1.14.0",
         "@svgr/webpack": "5.5.0",
+        "@types/aws-sdk": "^2.7.0",
```
The S3 und Mongo stuff should be removed for Lumi.

JPSchellenberg committed:
fix(deps): remove aws and mongodb dependencies

JPSchellenberg committed:
chore(cleanup): remove old integration

JPSchellenberg committed:
refactor(structure): add new client

JPSchellenberg committed:
test(lint): reinstall & run ts-lint

JPSchellenberg committed:
fix(client): long loading times when changing mode

JPSchellenberg committed:
fix(language): set language to en for now

JPSchellenberg committed:
fix(ui): change link color

JPSchellenberg committed:
fix(workingCache): delete h5p if tab is closed

JPSchellenberg committed:
Merge branch 'master' into update/server

JPSchellenberg committed:
Merge branch 'master' into update/server

JPSchellenberg committed:
Merge branch 'master' into update/server

JPSchellenberg commented:
Hey,

I refactored the app so that the client is built by the create-react-app toolchain and merged the electron and server parts.
What do you think?

sr258 commented:
App doesn't work if I have this env var: NODE_ENV=development Without it it starts fine.

JPSchellenberg commented:
Hmm, that's strange. Is PORT=8080?

sr258 commented:
Yes, the port is set to 8080

JPSchellenberg committed:
chore(deps): update package-lock.json

JPSchellenberg committed:
fix(api): make api naming consistent

JPSchellenberg committed:
fix(state): remove index files for export

JPSchellenberg committed:
fix(client): start with editor for now - dont show launchpad

JPSchellenberg committed:
fix(H5PEditor): remove useless constructor

JPSchellenberg committed:
refactor(state): merge h5p into h5peditor

JPSchellenberg committed:
fix(client): permanent loadingindicator while creating new h5p

JPSchellenberg committed:
fix(client): rename missing h5p to new h5p

JPSchellenberg committed:
Merge branch 'master' into refactor/app

JPSchellenberg committed:
test(client): remove deprecated test and install redux-mock-store

JPSchellenberg committed:
ci(install): run install in client

JPSchellenberg committed:
ci(build): remove deprecated version-info-script

JPSchellenberg committed:
fix(client): use correct api for opening files

JPSchellenberg committed:
test(spectron): remove deprecated jest-file and add spectron

JPSchellenberg committed:
refactor(client): add h5p-react instead of h5p-webcomponents (#1091)

* refactor(client): add h5p-react instead of h5p-webcomponents

* fix(h5p-react): corrected contentId for new content

Co-authored-by: Sebastian Rettig <serettig@posteo.de>

JPSchellenberg committed:
Merge branch 'master' into refactor/app

JPSchellenberg committed:
fix(deps): upgrade h5p-react to 0.1.1

JPSchellenberg committed:
refactor(h5peditor): remove deprecated code & use class-methods

JPSchellenberg committed:
fix(deps): upgrade h5p-nodejs-library to @lumieducation/h5p-server

JPSchellenberg committed:
chore(deps): reinstall husky & commitlint

JPSchellenberg committed:
chore(deps): pin dependencies

JPSchellenberg committed:
test(jest): add config

JPSchellenberg committed:
ci(npm): add npm token to build, test and release pipelines

JPSchellenberg committed:
chore(package): update author to GbR

JPSchellenberg committed:
ci(npm): replace workflow with echo script

JPSchellenberg committed:
ci(npm): add name to npm setup

JPSchellenberg committed:
ci(test): remove attach_workspace in test

JPSchellenberg committed:
fix(build-client): skip preflight check in create react app

The react-scripts package provided by Create React App requires a dependency:
  "jest": "26.6.0"
Don't try to install it manually: your package manager does it automatically.
However, a different version of jest was detected higher up in the tree:

JPSchellenberg committed:
ci(test-order): build server before client
